### PR TITLE
Add new menu entry to remove braces from selection, aka unprotect it.

### LINKED ISSUE #9950: Unprotect selection
When one right clicks on some selected text in a field, one has a choice of "Protect selection" (in submenu "Protect terms") that puts "{" before the selection and "}" after it. Perhaps it would be worth to have an opposite action, "Unprotect selection", that would remove all such "protections"? It would be especially convenient when, the imported record has protected words in all caps, and one wants to use, for example, title case.

### PR DESCRIPTION
**Pull request by BeeJay28**
Fixes #9950 

Add an additional right click menu entry to "unprotect text selection", which strips the selected text of any curly-brace pairs.

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable). (UnprotectTermsFormatterTest.java does the testing)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request. (No information there)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository. No documentation for this little right-click menu entry available. If an entry is necessary, then perhaps in general on how terms-protection works, what it is useful for and how to use it.

## UI Changes:
Unprotect with selected text
Unprotect with UNselected text

### CODE REVIEW

File: src/main/java/org/jabref/gui/fieldeditors/contextmenu/ProtectedTermsMenu.java [hunk 1]
```diff
@@ -54,6 +66,16 @@ public void execute() {
         }
     }
 
+    private class UnprotectSelectionAction extends SimpleCommand {
+
+        @Override
+        public void execute() {
+            String selectedText = textInputControl.getSelectedText();
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/fieldeditors/contextmenu/ProtectedTermsMenu.java:73]: Just use the UnprotectTermsFormatter

**Review by Siedlerchr — DISMISSED** [review]

**Comment by koppor** [comment]
The PR misses a screenshot:

image

It works there, but does not work if nothing is selected:

image

Three options:

a) gray out option if nothing is selected
b) Unprotect ALL if clicked
c) change text to "Unprotect" if nothing is selected and unprotect everything when clicked

**Review by koppor — CHANGES_REQUESTED** [review]
It does not behave properly when nothing is selected. See comments.

Minor: I had to fix a CHANGELOG merge error.

**Comment by BeeJay28** [comment]
> 
I was unsure of what it should exactly do if nothing is selected, so I just left it at the easiest option. Thanks for the feedback, I'll get on it.
And thank you for the screenshot. Tbh, I just had issues screenshotting while having the right click menu up (it would automatically close before capture). Thank you for that too 😅

**Comment by koppor** [comment]
> And thank you for the screenshot. Tbh, I just had issues screenshotting while having the right click menu up (it would automatically close before capture). Thank you for that too 😅

I am using Greenshot on Windows with Hotkey <kbd>Ctrl</kbd>+<kbd>F1</kbd> and that works most of the time. I also have experiences using GIMP, where one can say: Take a screenshot in 10 seconds. Then, one only needs "patience".

Thank you for continuing working on this!

**Comment by BeeJay28** [comment]
> Three options:
> 
> a) gray out option if nothing is selected b) Unprotect ALL if clicked c) change text to "Unprotect" if nothing is selected and unprotect everything when clicked

I picked option a)

**Review by koppor — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**